crates/modelardb_query/src/query/model_table.rs

```diff
@@ -56,17 +55,26 @@ use super::QUERY_ORDER_SEGMENT;
 /// containing metadata and models. [`ModelTable`] implements [`TableProvider`] so it can be
 /// registered with Apache DataFusion and the multivariate time series queried as multiple
 /// univariate time series.
-pub struct ModelTable {
-    /// Access to the system's configuration and components.
-    context: Arc<Context>,
-    /// Metadata required to read from and write to the model table.
+pub(crate) struct ModelTable {
+    /// Access to the Delta Lake table.
+    delta_table: DeltaTable,
+    /// Metadata for the model table.
     model_table_metadata: Arc<ModelTableMetadata>,
+    /// Were data should be written to.
+    data_sink: Arc<dyn DataSink>,
+    /// Aaccess to metadata related to tables.
+    table_metadata_manager: Arc<TableMetadataManager>,
     /// Field column to use for queries that do not include fields.
     fallback_field_column: u16,
 }
 
 impl ModelTable {
-    pub fn new(context: Arc<Context>, model_table_metadata: Arc<ModelTableMetadata>) -> Arc<Self> {
+    pub(crate) fn new(
+        delta_table: DeltaTable,
+        table_metadata_manager: Arc<TableMetadataManager>,
+        model_table_metadata: Arc<ModelTableMetadata>,
+        data_sink: Arc<dyn DataSink>,
```

Comment:
Is there a reason that this is `dyn DataSink` and not just `ModelTableDataSink`?

Reply:
If it is changed here, maybe also change it in the struct field type.

Reply:
It is `Arc<dyn DataSink>` as `ModelTableDataSink` uses `StorageEngine`. Thus, there seemed to be no way to use `ModelTableDataSink` here without `modelardb_query` having to depend on `modelardb_server`. Although there may be some way to do it that I did not realize. Another benefit is that with `data_sink` being `Arc<dyn DataSink>`, the model table can easily be configured to write data to other locations than `StorageEngine`.